src/util/makeEnterClickable.ts

```diff
@@ -1,69 +1,97 @@
 import { forceUpdate } from '@stencil/core';
 
-const eventHandlers: WeakMap<HTMLElement, CallBacks> = new WeakMap();
+const eventHandlers: WeakMap<HTMLElement, EnterClickable> = new WeakMap();
 
-export function makeEnterClickable(element: HTMLElement) {
-    if (eventHandlers.has(element)) {
-        return;
-    }
+class EnterClickable {
+    private isActive = false;
+    private hasJustReleasedEnter = true;
 
-    let isActive = false;
-    let hasJustReleasedEnter = true;
+    private handleKeyDown = (event: KeyboardEvent) => {
+        if (event.key === 'Enter' && !this.isActive) {
+            this.isActive = true;
 
-    const keydownHandler = (event: KeyboardEvent) => {
-        if (event.key === 'Enter' && !isActive) {
-            isActive = true;
-
-            forceUpdate(element);
+            if (this.element?.shadowRoot) {
+                forceUpdate(this.element);
+            }
         }
     };
 
-    const keyupHandler = (event: KeyboardEvent) => {
-        if (event.key === 'Enter' && isActive) {
-            isActive = false;
-            hasJustReleasedEnter = true;
+    private handleKeyUp = (event: KeyboardEvent) => {
+        if (event.key === 'Enter' && this.isActive) {
+            this.isActive = false;
+            this.hasJustReleasedEnter = true;
 
-            forceUpdate(element);
+            if (this.element?.shadowRoot) {
+                forceUpdate(this.element);
+            }
```

Comment:
It's a check to see if we are dealing with an `HTMLelement` generated from a Stencil component class. `forceUpdate` will throw an error if it's not a Stencil component.

But it might indeed fail in some weird cases, if for some reason we put this behavior on a non-stencil `HTMLelement` with a shadowRoot

Reply:
The fact that it's a shadowRoot does not mean that it's a stencil component? :)
https://developer.mozilla.org/en-US/docs/Web/API/Element/shadowRoot

Reply:
We found some strange property on the stencil components that we could use,`$hostElement,` but I think Johan advised against it. It can be seen here https://github.com/ionic-team/stencil/blob/4ce38c581d4b6d28ab9924bf8d6917b9801f62aa/src/runtime/update-component.ts#L362

During the bootstrap phase it calls `registerHost` which adds this property, so it seems all Stencil components have this property. So maybe we could use that?

Reply:
That would be nice but maybe we shouldn't use some internal attribute that might disappear in future releases. :(